server/background_tasks.go

```diff
@@ -73,13 +73,16 @@ func (s *Server) ListenForIntegrationsUpdateEvents() error {
 				s.Errorf("ListenForIntegrationsUpdateEvents: " + err.Error())
 				return
 			}
-			systemKeysCollection.UpdateOne(context.TODO(), bson.M{"key": "ui_url"},
-				bson.M{"$set": bson.M{"value": integrationUpdate.UIUrl}})
-			UI_url = integrationUpdate.UIUrl
 			switch strings.ToLower(integrationUpdate.Name) {
 			case "slack":
-				notifications.CacheSlackDetails(integrationUpdate.Keys, integrationUpdate.Properties)
+				systemKeysCollection.UpdateOne(context.TODO(), bson.M{"key": "ui_url"},
+					bson.M{"$set": bson.M{"value": integrationUpdate.UIUrl}})
+				UI_url = integrationUpdate.UIUrl
+				CacheDetails("slack", integrationUpdate.Keys, integrationUpdate.Properties)
+			case "s3":
+				CacheDetails("s3", integrationUpdate.Keys, integrationUpdate.Properties)
 			default:
+				s.Warnf("unkown integration")
```

Comment:
log must to contain the function name + add the wrong parameter + you have typo -- unknown